src/NexusMods.DataModel/ApplyService.cs

```diff
@@ -102,56 +85,49 @@ public ValueTask<FileDiffTree> GetApplyDiffTree(LoadoutId loadoutId)
         var lastAppliedLoadout = lastLoadout ?? throw new KeyNotFoundException("Loadout not found for last applied revision");
         
         var loadoutWithIngest = await lastAppliedLoadout.Ingest();
-        
-        // Get the latest loadout revision
-        var latestRevision = _loadoutRegistry.Get(loadoutWithIngest.LoadoutId);
-        if (latestRevision is null)
-            throw new KeyNotFoundException("No latest revision found for last applied loadout");
-        
-        // if latest revision is the same as the last applied revision, no need to rebase
-        if (latestRevision.DataStoreId.Equals(loadoutWithIngest.DataStoreId))
-        {
-            var newLoadout = _loadoutRegistry.Alter(loadoutWithIngest.LoadoutId, $"Ingested changes in loadout: {loadoutWithIngest.Name}",
-                l => loadoutWithIngest
-            );
-            return newLoadout;
-        }
-        
-        // Rebase unapplied changes on top of ingested changes
-        var mergedLoadout = _loadoutRegistry.Alter(latestRevision.LoadoutId, $"Rebase unapplied changes on top of ingested changes in loadout: {latestRevision.Name}",
-            l => l.Installation.GetGame().Synchronizer.MergeLoadouts(loadoutWithIngest, latestRevision)
-        );
-        
-        return mergedLoadout;
+
+        return loadoutWithIngest;
     }
 
     /// <inheritdoc />
-    public IId? GetLastAppliedLoadout(GameInstallation gameInstallation)
+    public Loadout.Model? GetLastAppliedLoadout(GameInstallation gameInstallation)
     {
-        var loadoutRevision = _diskStateRegistry.GetLastAppliedLoadout(gameInstallation);
-        return loadoutRevision;
+        if (!_diskStateRegistry.TryGetLastAppliedLoadout(gameInstallation, out var lastId))
+        {
+            return null;
+        }
+        
+        var db = _conn.AsOf(lastId.Tx);
+        return db.Get(lastId.Id);
     }
 
     /// <inheritdoc />
-    public IObservable<IId> LastAppliedRevisionFor(GameInstallation gameInstallation)
+    public IObservable<LoadoutRevisionId> LastAppliedRevisionFor(GameInstallation gameInstallation)
     {
+        LoadoutRevisionId last;
+        if (_diskStateRegistry.TryGetLastAppliedLoadout(gameInstallation, out var lastId))
+            last = lastId;
+        else
+            last = new LoadoutRevisionId(LoadoutId.From(EntityId.From(0)), TxId.From(0));
+        
         // Return a deferred observable that computes the starting value only on first subscription
         return Observable.Defer(() => _diskStateRegistry.LastAppliedRevisionObservable
-            .Where(x => x.gameInstallation.Equals(gameInstallation))
-            .Select(x => x.loadoutRevision)
-            .StartWith(_diskStateRegistry.GetLastAppliedLoadout(gameInstallation) ?? IdEmpty.Empty)
+            .Where(x => x.Install.Equals(gameInstallation))
+            .Select(x => x.LoadoutRevisionId)
+            .StartWith(last)
```

Comment:
Does this still have to be a deferred observable? @Al12rs do you have more info on this, I think you wrote this initially.

Reply:
The idea was to only start observing while there are actually interested parties. 
I think there might be better ways to do this but it should be fine for now.